libvast/vast/bits.hpp

```diff
@@ -99,6 +99,8 @@ class bits : detail::equality_comparable<bits<T>> {
   /// Checks whether all bits have the same value.
   /// @returns `true` if the bits are either all 0 or all 1.
   bool homogeneous() const {
+    if (is_run())
+      return true;
```

Comment:
This is a very performance-critical function. The extra branching should be justified. Have you checked that `all_or_none` (which computes `((x + 1) & lsb0) <= 1`) violate a precondition otherwise?

Reply:
We identified a more performant fix:
```
size_ >= word<T>::width
```